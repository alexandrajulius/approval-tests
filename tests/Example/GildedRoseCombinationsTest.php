<?php

declare(strict_types=1);

namespace AHJ\ApprovalTests\Tests\Example;

use AHJ\ApprovalTests\CombinationApprovals;
use PHPUnit\Framework\TestCase;

final class GildedRoseCombinationsTest extends TestCase
{
    public function testUpdateQualityWithCombinations(): void
    {
        $arguments = [
            ['foo', 'Aged Brie', 'zorg'],
            range(0, 5),
            [15, 8, 11],
        ];

        CombinationApprovals::create()->verifyAllCombinations(
            function (string $name, int $sellIn, int $quantity) {
                $items = [new Item($name, $sellIn, $quantity)];

                return (new GildedRose())->updateQuality($items);
            },
            $arguments
        );
    }
}
