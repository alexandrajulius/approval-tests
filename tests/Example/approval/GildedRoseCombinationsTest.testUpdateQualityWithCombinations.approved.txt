[Aged Brie, 3, 15] -> [Aged Brie, 2, 16]
[foo, 3, 15] -> [foo, 2, 14]